packages/app-contracts/src/Codes/Upload.tsx

```diff
@@ -0,0 +1,172 @@
+// Copyright 2017-2019 @polkadot/app-contracts authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { SubmittableResult } from '@polkadot/api/SubmittableExtrinsic';
+
+import BN from 'bn.js';
+import React from 'react';
+import { Button, InputFile, TxButton } from '@polkadot/ui-app';
+import { compactAddLength } from '@polkadot/util';
+import { Hash } from '@polkadot/types';
+
+import ContractModal, { ContractModalProps, ContractModalState } from '../Modal';
+import store from '../store';
+import translate from '../translate';
+
+type Props = ContractModalProps;
+
+type State = ContractModalState & {
+  gasLimit: BN,
+  isWasmValid: boolean,
+  wasm?: Uint8Array | null
+};
+
+class Upload extends ContractModal<Props, State> {
+  constructor (props: Props) {
+    super(props);
+    this.defaultState = {
```

Comment:
Same comments around `state` and `defaultState` explicit type setting (as per previous class)